src/main/java/org/repositoryminer/postmining/DefectMetricsPostMiningTask.java

```diff
@@ -0,0 +1,186 @@
+ * We thus measure the defect-proneness of a class c in version v using the following three different ways, respectively: 
+ * 1. The number of times class c is involved in fixing bugs that were injected in version v, that is, the version where the bugs were found and reported.
+ * 2. The number of times class c is involved in fixing bugs that were resolved in version v.
+ * 3. The number of times class c is involved in fixing bugs that were alive in version v; 
+ *   that is, the bugs were reported before or in version v and were resolved after or in version v.
+ */
+public class DefectMetricsPostMiningTask implements IPostMiningTask{
+
+	ReferenceDocumentHandler referenceDH = new ReferenceDocumentHandler();
+	CommitDocumentHandler commitDH = new CommitDocumentHandler();
+	IssueDocumentHandler issueDH = new IssueDocumentHandler();
+	
+	@Override
+	public String getTaskName() {
+		return "Defect-proneness measurement";
+	}
+
+	@Override
+	public void execute(RepositoryMiner miner, Repository repository, IPostMiningListener listener) {
+					
+		//Define start and end dates for each version
+		List<Document> versionList =  referenceDH.findAll();
+		List<Reference> tagsList = Reference.parseDocuments(versionList);
+		HashMap<String, Date> tags = defineVersionPeriod(versionList);
+		List<Document>issuesList =  issueDH.findAll();
+		
+		//Link closed issues to each version
+		HashMap<String, List<Commit>> issuesOpen = new HashMap<String, List<Commit>>();
+		HashMap<String, List<Commit>> issuesClosed = new HashMap<String, List<Commit>>();
+		HashMap<String, List<Commit>> issuesAlive = new HashMap<String, List<Commit>>();
+		
+		for (Reference reference : tagsList) {
+			issuesOpen.put(reference.getId(), new ArrayList<Commit>());
+			issuesClosed.put(reference.getId(), new ArrayList<Commit>());
+			issuesAlive.put(reference.getId(), new ArrayList<Commit>());					
+		}		
+		if (issuesList != null && !issuesList.isEmpty()) {
+			for (Document document : issuesList) {					
+				Date openDate = document.getDate("created_at");
+				Date closeDate = document.getDate("closed_at");
+				if (isBug(document)) {
+					List<Document> events = (List<Document>) document.get("events");
+					for (Document event : events) {
+						if (event.getString("commit") != null) {							
+							Document commitDoc = commitDH.findById(event.getString("commit"));
+							Commit commit = new Commit();
+							try {
+								if (commitDoc != null)
+									commit = Commit.parseDocument(commitDoc);
+							} catch (Exception e) {
+								System.out.println(e.getStackTrace());
+							}
+							
+							Boolean isOpenBefore = false;
+							Boolean isClosedBefore = false;
+							for (Reference reference : tagsList) {
+								Boolean isClosedSameTag = false;
+								if (openDate != null && tags.get(reference.getId()).after(openDate) && !isOpenBefore) {

```

Comment:
IMO `reference.getId()` should be extracted into a variable because it appears multiple times here.